README.txt

Segue v. 2.2.10.2 (2009-06-08)
=================================

What is Segue?
------------------
Segue is an open source collaborative content management system designed for
e-learning that combines the ease of use of course management systems with the
flexibility of weblogs for creating various types of sites including course, news,
and journal.


Current Version Notes
---------------------
This release of Segue fixes a bug in the standard course template. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.


Downloads
---------------------
For the latest and archived versions, please download from SourceForge.

http://sourceforge.net/project/showfiles.php?group_id=82171&package_id=246956


Documentation
---------------------
Segue includes contextual help for users. Additional documentation can be found
online at:

http://sourceforge.net/project/showfiles.php?group_id=82171&package_id=246956


Installation
---------------------
See the INSTALL.txt file in the Segue root directory for installation instructions
or read on the web at:

http://sourceforge.net/project/showfiles.php?group_id=82171&package_id=246956


Bug Tracker
---------------------
http://sourceforge.net/tracker/?group_id=82171&atid=565234







===================================================================
| Prior Segue Release Notes
| (See the Segue change log for more details)
===================================================================


v. 2.2.10.2 (2009-06-08)
----------------------------------------------------
This release of Segue fixes a bug in the standard course template. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.



v. 2.2.10.1 (2009-06-08)
----------------------------------------------------
This release of Segue includes refinements to the roles UI, participation plugin,
and adds a number of new site templates. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.



v. 2.2.9.3 (2009-03-20)
----------------------------------------------------
This release of Segue updates the Segue v1 to v2 migration code to support the
migration of v1 file metadata 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.



v. 2.2.9.2 (2009-02-26)
----------------------------------------------------
This release of Segue adds a new "Site-Members" feature. Every site has a
"Site-Members" group that can be used for setting the roles on many users at once.
It can also be used in conjunction with the new "Join Site" plugin to enable
self-registration (with moderation) to the site. This release also fixes a large
number of bugs. See the change log for more details. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.



v. 2.2.9.1 (2009-02-25)
----------------------------------------------------
This release of Segue adds a new "Site-Members" feature. Every site has a
"Site-Members" group that can be used for setting the roles on many users at once.
It can also be used in conjunction with the new "Join Site" plugin to enable
self-registration (with moderation) to the site. This release also fixes a large
number of bugs. See the change log for more details. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.



v. 2.2.9.0 (2009-02-25)
----------------------------------------------------
This release of Segue adds a new "Site-Members" feature. Every site has a
"Site-Members" group that can be used for setting the roles on many users at once.
It can also be used in conjunction with the new "Join Site" plugin to enable
self-registration (with moderation) to the site. This release also fixes a large
number of bugs. See the change log for more details. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.



v. 2.2.8.1 (2009-02-10)
----------------------------------------------------
This release also fixes a number of of minor bugs. See the change-log for details. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.4 and Polyphony 1.5.1.



v. 2.2.8.0 (2009-02-09)
----------------------------------------------------
This release adds a new tracking module to allow users to see their own actions
within a site and editors to see all of the actions within a site.

The Tags plugin has been significantly improved to allow choosing of the part of
the site from which tags are displayed. This could be the entire site or just a
section or page. The Tags plugin also gains display-options for ordering,
display-style (cloud/list), and number of tags to show.

This release also fixes a number of additional minor bugs. See the change-log for
details. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.4 and Polyphony 1.5.0.



v. 2.2.7.2 (2009-02-03)
----------------------------------------------------
This release fixes a number of issues with editing in Internet Explorer 7. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.3 and Polyphony 1.4.11.



v. 2.2.7.1 (2009-02-02)
----------------------------------------------------
This release fixes a few minor issues with the media library and improves support
for Middlebury's MiddMedia service.

See the change log for more details. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.3 and Polyphony 1.4.11.



v. 2.2.7.0 (2009-01-28)
----------------------------------------------------
This release adds support for external media libraries with support for Middlebury's
new MiddMedia service included. A new 'audio' text-template is included for
embedding an MP3 player in TextAndImages blocks and text-areas now get an 'Embed
Audio/Video' button. When migrating sites from Segue1 installations, mp3 downloads
are now converted to AudioPlayer plugins. The Portal styling has been updated to
better highlight the selected folder.

See the change log for more details. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.2 and Polyphony 1.4.10.



v. 2.2.6.0 (2009-01-12)
----------------------------------------------------
This release adds support for creating new content through wiki-links. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.8.6 and Polyphony 1.4.9.



v. 2.2.5.7 (2009-01-12)
----------------------------------------------------
This release fixes a character-encoding issue in which some Russian text became
garbled due to incorrect encoding conversion. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.8.6 and Polyphony 1.4.9.



v. 2.2.5.6 (2009-01-09)
----------------------------------------------------
This release fixes a number of bugs found by users over the past few months,
improving migration, the audio plugin, documentation, RSS display, and a few other
issues. See the change-log for details. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.8.5 and Polyphony 1.4.9.



v. 2.2.5.5 (2008-10-22)
----------------------------------------------------
This release add the ability to make empty site-placeholders aliases to placeholders
that contain sites. This enables sites to be moved and have users redirected from
the old urls. As well making multiple course-section-placeholders aliases to a
single site for the course can allow that site to show up under the 'current
classes' listing for students in all sections of the course.

The user interface for copying sites has now been merged with the user-interface
for copying content for a more consistent user experience.

The 2.2.5.5 bug-fix release adds further fixes authenticated RSS feeds

See the change-log for details on additional fixes and improvements. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.8.4 and Polyphony 1.4.8.



v. 2.2.5.4 (2008-10-16)
----------------------------------------------------
This release add the ability to make empty site-placeholders aliases to placeholders
that contain sites. This enables sites to be moved and have users redirected from
the old urls. As well making multiple course-section-placeholders aliases to a
single site for the course can allow that site to show up under the 'current
classes' listing for students in all sections of the course.

The user interface for copying sites has now been merged with the user-interface
for copying content for a more consistent user experience.

The 2.2.5.4 bug-fix release adds further fixes upgrade mechanisms.

See the change-log for details on additional fixes and improvements. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.8.3 and Polyphony 1.4.8.



v. 2.2.5.3 (2008-10-16)
----------------------------------------------------
This release add the ability to make empty site-placeholders aliases to placeholders
that contain sites. This enables sites to be moved and have users redirected from
the old urls. As well making multiple course-section-placeholders aliases to a
single site for the course can allow that site to show up under the 'current
classes' listing for students in all sections of the course.

The user interface for copying sites has now been merged with the user-interface
for copying content for a more consistent user experience.

The 2.2.5.2 bug-fix release adds further fixes to the attribution display.

See the change-log for details on additional fixes and improvements. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.8.3 and Polyphony 1.4.8.



v. 2.2.5.2 (2008-10-15)
----------------------------------------------------
This release add the ability to make empty site-placeholders aliases to placeholders
that contain sites. This enables sites to be moved and have users redirected from
the old urls. As well making multiple course-section-placeholders aliases to a
single site for the course can allow that site to show up under the 'current
classes' listing for students in all sections of the course.

The user interface for copying sites has now been merged with the user-interface
for copying content for a more consistent user experience.

The 2.2.5.2 bug-fix release adds logging to alias actions and fixes to the
tag-viewing actions.

See the change-log for details on additional fixes and improvements. 

----

 Upgrades from versions prior to 2.2.5.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.8.3 and Polyphony 1.4.8.



v. 2.2.5.1 (2008-10-14)
----------------------------------------------------
This release add the ability to make empty site-placeholders aliases to placeholders
that contain sites. This enables sites to be moved and have users redirected from
the old urls. As well making multiple course-section-placeholders aliases to a
single site for the course can allow that site to show up under the 'current
classes' listing for students in all sections of the course.

The user interface for copying sites has now been merged with the user-interface
for copying content for a more consistent user experience.

The 2.2.5.1 bug-fix release adds improvements to the installation process to help
ensure that new installations are operational without running database updates.

See the change-log for details on additional fixes and improvements.

Upgrades from versions prior to 2.2.5.0 require running system updaters located
under Admin Tools --> Segue Updates.

This Segue release uses Harmoni 1.8.2 and Polyphony 1.4.7.



v. 2.2.5.0 (2008-10-13)
----------------------------------------------------
This release add the ability to make empty site-placeholders aliases to placeholders
that contain sites. This enables sites to be moved and have users redirected from
the old urls. As well making multiple course-section-placeholders aliases to a
single site for the course can allow that site to show up under the 'current
classes' listing for students in all sections of the course.

The user interface for copying sites has now been merged with the user-interface
for copying content for a more consistent user experience.

See the change-log for details on additional fixes and improvements.

Upgrades from versions prior to 2.2.5.0 require running system updaters located
under Admin Tools --> Segue Updates.

This Segue release uses Harmoni 1.8.2 and Polyphony 1.4.7.



v. 2.2.4.1 (2008-10-03)
----------------------------------------------------
This bug-fix release includes a number of small usability fixes to improve the user
experience. See the change-log for details.

Upgrades from versions prior to 2.2.4.0 require running system updaters located
under Admin Tools --> Segue Updates.

This Segue release uses Harmoni 1.8.1 and Polyphony 1.4.6.



v. 2.2.4.0 (2008-09-24)
----------------------------------------------------
This release includes a large number of usability improvements that make Segue usage
much more smooth.

Notable User-Facing changes:

* New 'Recently Visited' folder

 * The current portal folder and many settings are now remembered.

 * New media library buttons in the Text and Images plugin make it much easier to
insert images and add links to files.

 * Permissions icons are now shown in edit-mode and in the media library.

 * The move/copy process has be renamed to better fit with the copy/paste metaphor
of usage.

 * The selection panel now is always shown.

 * The window now scrolls when dragging in Arrange mode.

See the change log for more details and other improvements.

Upgrades from versions prior to 2.2.4.0 require running system updaters located
under Admin Tools --> Segue Updates. 

----

 This Segue 2 release uses Harmoni 1.8.0 and Polyphony 1.4.5.



v. 2.2.3 (2008-09-17)
----------------------------------------------------
This release fixes a number of small bugs and usability issues. 

----

 This Segue 2 release uses Harmoni 1.7.0 and Polyphony 1.4.4.



v. 2.2.2 (2008-08-29)
----------------------------------------------------
This release fixes several issues that were in 2.2.1. 

----

 This Segue 2 release uses Harmoni 1.7.0 and Polyphony 1.4.3.



v. 2.2.1 (2008-08-29)
----------------------------------------------------
This release improves the speed of searching for sites in the portal as well as
fixes a number of minor bugs.

Notable user-facing changes:

 * speed improvements in the portal

 * Audio Clip plugin can now be used in comments.

See the change log for more details on these and other fixes. 

----

 This Segue 2 release uses Harmoni 1.7.0 and Polyphony 1.4.3.



v. 2.2.0 (2008-08-27)
----------------------------------------------------
This Release adds several new plugins, a number of user-interface and RSS
improvements, and numerous fixes.

Notable User-facing changes:

* New Audio Clip plugin

 * RSS feeds now include enclosures for File for Download and Audio Clip plugins.

 * RSS feeds now include tags.

See the change log for more details on these and other fixes. 

----

 This Segue 2 release uses Harmoni 1.7.0 and Polyphony 1.4.3.



v. 2.1.0 (2008-08-25)
----------------------------------------------------
This release updates site URLs to include the site/placeholder name to improve the
sense of context by users. Additionally, this release includes 'Meebo' chat widget
support as well as a number of fixes.

See the change-log for details of these fixes. 

----

 This Segue 2 release uses Harmoni 1.6.2 and Polyphony 1.4.2.



v. 2.0.1 (2008-08-21)
----------------------------------------------------
This bug fix release fixes an error in editing multi-lingual text in 'new mode' as
well as a minor Segue 1 redirect issue.

See the change-log for details of these fixes. 

----

 This Segue 2 release uses Harmoni 1.6.1 and Polyphony 1.4.1.



v. 2.0.0 (2008-08-20)
----------------------------------------------------
This first stable release of Segue 2 fixes a number of issues that were found during
testing of the release candidates. 

Notable user-facing changes:

* Now supports embedded GoogleCalendars.

 * Now supports links targeting pop-up windows.

 * YouTube videos now can be loaded in high quality using the fmt=18 option.

 * A warning is now given if leaving a page with a text-editor open.

See the change-log for details of these and other fixes. 

----

 Upgrades:

As of beta 25, Segue requires PHP to be compiled with XSL support. On unix systems
this can be added by configuring PHP using the --with-xsl option. See the following
for installation instructions: http://www.php.net/manual/en/xsl.installation.php

Upgrades from versions prior to beta 30 require running system updaters located
under Admin Tools --> Segue Updates. 

----

 This Segue 2 release uses Harmoni 1.6.1 and Polyphony 1.4.1.



v. 2.0-rc-2 (2008-08-14)
----------------------------------------------------
This release fixes a number of small bugs and a number of security issues.

Security issues fixed: 

* Cross-Site Request Forgeries (CSRF) are now eliminated from data-modification
actions. Read about CSRF at: http://shiflett.org/articles/cross-site-request-forgeries

 * Segue can now be configured to use secure/http-only cookies. 

 * Admin actions are now restricted to prevent listing of users and ids in the
system. 



v. 2.0-rc-1 (2008-08-14)
----------------------------------------------------
This first production release candidate of Segue 2 fixes a number of user-interface
difficulties and adds much documentation.

Notable User-Facing Changes:

* Can now search for sites in the Portal by title, description, or placeholder

 * More contextual help.

 * New 'Blank' template.

 * Can now set site-wide roles and theme when creating a site.

 * Roles links have been moved to the top of the screen for easier access. 

----

 Upgrades:

As of beta 25, Segue requires PHP to be compiled with XSL support. On unix systems
this can be added by configuring PHP using the --with-xsl option. See the following
for installation instructions: http://www.php.net/manual/en/xsl.installation.php

Upgrades from versions prior to beta 30 require running system updaters located
under Admin Tools --> Segue Updates. 

----

 Segue 2.0.0 uses Harmoni 1.5.0 and Polyphony 1.3.3.



v. 2.0-beta-30 (2008-08-07)
----------------------------------------------------
This release fixes a number of security issues, an authorization bug that was
prevent proper view access to some sites, and a few smaller issues. As well, it
introduces the ability to move or copy content and pages within a site or across
sites. See the change-log for details.

Notable User-Facing Changes:

* New Move/Copy abilities

 * New delete-confirmation dialog that shows the scope of the action. 

----

 Upgrades:

As of beta 25, Segue requires PHP to be compiled with XSL support. On unix systems
this can be added by configuring PHP using the --with-xsl option. See the following
for installation instructions: http://www.php.net/manual/en/xsl.installation.php

Upgrades from versions prior to beta 30 require running system updaters located
under Admin Tools --> Segue Updates. 

----

 Segue 2.0-beta-30 uses Harmoni 1.4.7 and Polyphony 1.3.2.



v. 2.0-beta-29 (2008-08-01)
----------------------------------------------------
This release adds a number of new themes, allows copying of sites, and numerous
other improvements.

Notable User-facing changes:

* Can now copy sites to empty placeholders in the portal.

 * Site-URLs are now printed in the portal.

 * Viewing of sites has now been dramatically sped-up for users without the editer role.

 * Sections and Pages are now named with more clarity.

 * Many theme improvements.

 * Portal style-fixes MS Internet Explorer browser.

 * Better support for uploading multi-page PDFs.

See the change log for more details and other fixes.

Harmoni 1.4.6 includes a fix to the thumbnail generation system that prevents
excess processing overhead on large multi-page PDFs and improves the upload and
migration of these files.

Segue 2.0-beta-28 uses Harmoni 1.4.6 and Polyphony 1.3.1. 

----

 Upgrades:

As of this beta 25, Segue requires PHP to be compiled with XSL support. On unix
systems this can be added by configuring PHP using the --with-xsl option. See the
following for installation instructions: http://www.php.net/manual/en/xsl.installation.php

Upgrades from versions prior to beta 26 require running the RSS Feed updaters
located under Admin Tools --> Segue Updates.



v. 2.0-beta-28 (2008-07-24)
----------------------------------------------------
This release fixes a number of minor bugs, improves theme-support for MS Internet
Explorer, and adds support for embedding GoogleMaps. 

See the change-log for additional details on fixes.

Segue 2.0-beta-28 uses Harmoni 1.4.5 and Polyphony 1.3.0. 

----

 Upgrades:

As of this beta 25, Segue requires PHP to be compiled with XSL support. On unix
systems this can be added by configuring PHP using the --with-xsl option. See the
following for installation instructions: http://www.php.net/manual/en/xsl.installation.php

Upgrades from versions prior to beta 26 require running the RSS Feed updaters
located under Admin Tools --> Segue Updates.



v. 2.0-beta-27.2 (2008-07-21)
----------------------------------------------------
This release fixes a number of bugs in migration and with wiki-text.

An additional bug fix in Harmoni 1.4.4 fixes an issue with permissions settings for
members of groups the ids of which contain quotes.

See the change-log for additional details on fixes.

Segue 2.0-beta-27 uses Harmoni 1.4.4 and Polyphony 1.2.11. 

----

 Upgrades:

As of this beta 25, Segue requires PHP to be compiled with XSL support. On unix
systems this can be added by configuring PHP using the --with-xsl option. See the
following for installation instructions: http://www.php.net/manual/en/xsl.installation.php

Upgrades from versions prior to beta 26 require running the RSS Feed updaters
located under Admin Tools --> Segue Updates.



v. 2.0-beta-27.1 (2008-07-18)
----------------------------------------------------
This release fixes an issue that preventing inline images from appearing in text
blocks. 



v. 2.0-beta-27 (2008-07-17)
----------------------------------------------------
This release adds support for embedded Flash videos, text-templates, and fixes
several bugs.

Text-templates are a new extension system that use the same markup as MediaWiki's
templates to enable output of custom HTML blocks customized with a set of named
parameters. 

For example, a new 'video' template inserts embed code for videos from a variety of
providers defined in a white-list (to prevent XSS attacks). Example of usage:
{{video|service=youtube|id=s13dLaTIHSg|width=425|height=344}} This video template
will also search through the HTML source in a text-block and replace any embed
orobject elements with appropriate template markup.

Also included is a 'search' template that can insert a customized search field into
a text-block. Example of usage: {{search}} Example of usage: {{search|base_url=http://youtube.com/results|search_param_name=search_query}}

See the change-log for additional details on fixes.

Segue 2.0-beta-27 uses Harmoni 1.4.3 and Polyphony 1.2.10. 

----

 Upgrades:

As of this beta 25, Segue requires PHP to be compiled with XSL support. On unix

doc/changelog.txt

Segue Change Log


v. 2.2.11.0 (2010-03-04)
----------------------------------------------------
This release adds support for CAS single-sign-on authentication. 

----

 An updater located under Admin Tools --> Segue Updates can be used to mapp
accounts from existing authentication systems to CAS. 

----

 This Segue release uses Harmoni 1.10.1 and Polyphony 1.5.5.
		
		
	* New feature: 
		Added updater for mapping existing user accounts to CAS user accounts.
		(2010-03-04 - Adam Franco)
		
	* New feature: 
		Added support for CAS authentication.
		(2010-03-04 - Adam Franco)
		
	* New feature: 
		Added configuration lines to work with polyphony 1.5.3.
		(2010-03-04 - Adam Franco)
		
	* New feature: 
		Added a footer to allow the addition of statistics tracking scripts.
		(2010-03-04 - Adam Franco)
	



v. 2.2.10.3 (2009-09-30)
----------------------------------------------------
This release adds support for site-wide footers. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.7 and Polyphony 1.5.2.
		
		
	* New feature: 
		Now can add a site-wide footer for tracking codes.
		(2009-09-30 - Adam Franco)
		(reported by achapin)
		
	* Bug Fix: 
		CourseManagement: Update to course format for Midd's new LDAP class-group
		structure. 
		(2009-09-30 - Adam Franco)
		
	* Bug Fix: 
		Participation: Now doesn't die when a group is an administrator.
		(2009-09-30 - Adam Franco)
		
	* Bug Fix: 
		Participation: now doesn't die if no display name is available.
		(2009-09-30 - Adam Franco)
		
	* Bug Fix: 
		Themes: added getExtension() method to theme images in preparation for
		harmoni-1.10.0. 
		(2009-09-30 - Adam Franco)
	



v. 2.2.10.2 (2009-06-08)
----------------------------------------------------
This release of Segue fixes a bug in the standard course template. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.
		
		
	* Bug Fix: 
		site-templates: Fixed bug in new standard course site-templates.
		(2009-06-07 - Alex Chapin)
	



v. 2.2.10.1 (2009-06-08)
----------------------------------------------------
This release of Segue includes refinements to the roles UI, participation plugin,
and adds a number of new site templates. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.
		
		
	* Change: 
		site-templates: Created new site-templates including a standard course
		template in English, Arabic, Chinese, Japanese, French, Spanish, German and
		Portuguese. 
		(2009-06-07 - Alex Chapin)		
		
	* Change: 
		Audio and File plugins: these plugins now process and display text-templates
		in their captions.
		(2009-06-07 - Alex Chapin)		
		
	* Change: 
		Participation plugin: plugin now displays list of Site Administrators.
		(2009-04-15 - Alex Chapin)
		
	* Change: 
		Roles UI: Site members now listed in initial roles UI
		(2009-04-15 - Alex Chapin)
		
	* Change: 
		Participation plugin: added option to show or hide "other participants"
		(i.e. those participants who are not site members.
		(2009-04-15 - Alex Chapin)
		
	* Change: 
		tags display: changed how tagged items are displayed so is more flexible and
		works with various site layouts.
		(2009-04-15 - Alex Chapin)
		(reported by Bryan Carson of Middlebury College)
	



v. 2.2.9.3 (2009-03-20)
----------------------------------------------------
This release of Segue updates the Segue v1 to v2 migration code to support the
migration of v1 file metadata 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.
		
		
	* Bug Fix: #2692537 
		Migration: file metadata is now migration from v1 to v2.
		(2009-03-18 - Alex Chapin)
		(reported by Dick Forman of Middlebury College)
		
	* Change: 
		Migration:v1 participant lists now being imported at participation plugins.
		(2009-03-17 - Alex Chapin)
	



v. 2.2.9.2 (2009-02-26)
----------------------------------------------------
This release of Segue adds a new "Site-Members" feature. Every site has a
"Site-Members" group that can be used for setting the roles on many users at once.
It can also be used in conjunction with the new "Join Site" plugin to enable
self-registration (with moderation) to the site. This release also fixes a large
number of bugs. See the change log for more details. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.
		
		
	* Change: 
		Help: added a number of "Help" links to the Site-Membership and Join Site
		interfaces. 
		(2009-02-26 - Adam Franco)
	



v. 2.2.9.1 (2009-02-25)
----------------------------------------------------
This release of Segue adds a new "Site-Members" feature. Every site has a
"Site-Members" group that can be used for setting the roles on many users at once.
It can also be used in conjunction with the new "Join Site" plugin to enable
self-registration (with moderation) to the site. This release also fixes a large
number of bugs. See the change log for more details. 

----

 Upgrades from versions prior to 2.2.9.0 require running system updaters
located under Admin Tools --> Segue Updates. 

----

 This Segue release uses Harmoni 1.9.5 and Polyphony 1.5.2.
		
		
	* Bug Fix: 
		Plugins: getLocalAction() now returns the proper string.
		(2009-02-25 - Adam Franco)
		(reported by Alex Chapin of Middlebury College)
	



v. 2.2.9.0 (2009-02-25)
----------------------------------------------------
This release of Segue adds a new "Site-Members" feature. Every site has a

main/include/setup.inc.php
<?php
/**
 * This is the main control script for the application.
 *
 * @package segue
 * 
 * @copyright Copyright &copy; 2005, Middlebury College
 * @license http://www.gnu.org/copyleft/gpl.html GNU General Public License (GPL)
 *
 * @version $Id: setup.inc.php,v 1.11 2008/04/04 20:23:14 achapin Exp $
 */

/*********************************************************
 * Set a low debug level to not store up all queries.
 *********************************************************/
debug::level(-100);

/******************************************************************************
 * Start the session so that we can use the session for storage.
 ******************************************************************************/
if (file_exists(MYDIR.'/config/harmoni.conf.php'))
	require_once (MYDIR.'/config/harmoni.conf.php');
else
	require_once (MYDIR.'/config/harmoni_default.conf.php');
	
if (file_exists(MYDIR.'/config/action.conf.php'))
	require_once (MYDIR.'/config/action.conf.php');
else
	require_once (MYDIR.'/config/action_default.conf.php');
 
$harmoni->startSession();


/*********************************************************
 * If we pressed a button to reset segue, clear the session
 * and delete our tables.
 *********************************************************/
if (file_exists(MYDIR.'/config/debug.conf.php'))
	require_once (MYDIR.'/config/debug.conf.php');
else
	require_once (MYDIR.'/config/debug_default.conf.php');
		
if (isset($_REQUEST["reset_segue"]) 
	&& defined('ENABLE_RESET') 
	&& ENABLE_RESET) 
{
	$_SESSION = array();
	if (file_exists(MYDIR.'/config/database.conf.php'))
		require_once (MYDIR.'/config/database.conf.php');
	else
		require_once (MYDIR.'/config/database_default.conf.php');
	
	$dbc = Services::getService("DatabaseManager");
	$tableList = $dbc->getTableList($dbID);
	if (count($tableList)) {
		$queryString = "DROP TABLE `".implode("`, `", $tableList)."`;";
		print $queryString;
		$query = new GenericSQLQuery($queryString);
		$dbc->query($query, $dbID);
	}
}

/******************************************************************************
 * Include our configs
 ******************************************************************************/
require_once(HARMONI."/oki2/shared/ConfigurationProperties.class.php");
require_once(OKI2."/osid/OsidContext.php");

$configs = array(	
					'validation',
					'debug',
					'starting_site',
					'harmoni',
					'action',
					'database',
					'id',
					'logging',
					'recaptcha',
					'authentication_setup',
					'gui',
					'language',
					'help',
					'sets',
					'mime',
					'imageprocessor',
					'hierarchy',
					'authorization',
					'installer',
					'agent',
					'datamanager',
					'repository',
					'plugins',
					'post_config_setup',
					'viewer',
					'slots',
					'scheduling',
					'grading',
					'coursemanagement',
					'dataport',
					'tagging',
					'templates',
					'themes',
					'uploads',
					'welcome',
					'media_libraries'
				);

foreach ($configs as $config) {
	if (file_exists(MYDIR.'/config/'.$config.'.conf.php'))
		require_once (MYDIR.'/config/'.$config.'.conf.php');
	else
		require_once (MYDIR.'/config/'.$config.'_default.conf.php');
}

/*********************************************************
 * Set our starting site if needed.
 *********************************************************/
if (defined('SEGUE_STARTING_SITE') && SEGUE_STARTING_SITE) {
	$harmoni->config->set("defaultModule","view");
	$harmoni->config->set("defaultAction","html");
	$harmoni->config->set("defaultParams",array("site" => SEGUE_STARTING_SITE));
}

/*********************************************************
 * Set a list of actions that require request tokens to prevent 
 * Cross-Site Request Forgery attacks. All actions that 
 * could potentially change data should require this.
 *
 * Actions in this list will not be able to be loaded directly.
 *********************************************************/
$harmoni->ActionHandler->addRequestTokenRequiredActions(array(
		"comments.*",
		"dataport.convert",
		"dataport.import",
		"media.delete",
		"media.update",
		"media.upload",
		"plugin_manager.update_ajax",
		"media.delete",
		"portal.copy_site",
		"roles.choose_agent",
		"roles.modify",
		"roles.rebuildImplicit",
		"slots.delete",
		"slots.edit",
		"slots.make_alias",
		"slots.remove_alias",
		"ui1.add",
		"ui1.add_wiki_component",
		"ui1.addContent",
		"ui1.addMenuContent",
		"ui1.deleteComponent",
		"ui1.editContent",
		"ui1.editFlowOrg",
		"ui1.editHeader",
		"ui1.editMenu",
		"ui1.editNav",
		"ui1.editSite",
		"ui1.reorder",
		"ui1.theme_options",
		"ui2.add",
		"ui2.add_wiki_component",
		"ui2.addComponent",
		"ui2.createSubMenu",
		"ui2.deleteComponent",
		"ui2.modifyComponent",
		"ui2.moveComponent",
		"ui2.reorder",
		"updates.*",
		"versioning.revert"
	));
